hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopValidator/PigValidator/PigDependencyValidator.groovy

```diff
@@ -0,0 +1,391 @@
+
+
+import com.linkedin.gradle.hadoopdsl.NamedScope;
+import com.linkedin.gradle.hadoopdsl.job.PigJob;
+import com.linkedin.gradle.hdfs.HdfsFileSystem;
+import org.apache.hadoop.fs.Path;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.TaskAction;
+
+import java.nio.file.Paths;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * PigDependencyValidator is the class that provides the Task for validation of dependencies mentioned in the Apache Pig Scripts
+ * in the project.
+ */
+class PigDependencyValidator extends DefaultTask implements PigValidator {
+
+  static final int LOCAL_REL = 0, LOCAL_ABS = 1, HDFS = 2, REPO = 3;
+
+  Map<PigJob, NamedScope> jobMap;
+  Properties properties;
+  def fileSystem;
+  boolean error = 0;
+  ArrayList<Tuple> err_paths = null;
+  ArrayList<File> zipContents;
+  String libpath;
+
+  /**
+   * Validates the dependencies mentioned in the Apache Pig Scripts
+   * in the project. This is the Task Action Function
+   */
+  @TaskAction
+  void validate() {
+    String path;
+
+    //project.dependencies
+    ArrayList<Dependency> udfs = new ArrayList<Dependency>();
+    File script;
+    if (err_paths == null) {
+      err_paths = new ArrayList<Tuple>();
+    }
+    err_paths.clear();
+    error = false;
+
+    InputStream krbInputStream = this.getClass().getClassLoader().getResourceAsStream("oozie-krb5.conf");
+    File krb5 = new File(System.getProperty("java.io.tmpdir"), "krb5.conf");
+    OutputStream krbOutputStream = new FileOutputStream(krb5);
+
+    int read;
+    byte[] bytes = new byte[1024];
+    while ((read = krbInputStream.read(bytes)) != -1) {
+      krbOutputStream.write(bytes, 0, read);
+    }
+
+    URI clusterURI = null;
+    String clusterURIString;
+    String repo_url = null;
+    String fileText;
+    String alt_reg_command;
+    Matcher matcher;
+
+    jobMap.each { PigJob pigJob, NamedScope parentScope ->
+      script = new File(pigJob.script);
+      if (script.name.endsWith(".pig")) {
+        project.logger.lifecycle("Checking file: \t $script");
+        File subst_file = new File("${script}.substituted");
+        udfs = extractDependencies(subst_file);
+        fileText = subst_file.text;
+        udfs.each { dep ->
+          path = dep.filename;
+          if (dep.pathType == Dependency.PathType.HDFS) {
+            path = getPath(path);
+            if (clusterURI == null) {
+              initHdfsFileSystem(krb5);
+              if (properties.containsKey("NameNode")) {
+                clusterURIString = properties.getProperty("NameNode");
+              } else {
+                properties.each { key, value ->
+                }
+                throw new GradleException(
+                    "Please specify NameNode address in ${project.projectDir}/.hadoopValidatorProperties");
+              }
+              if (!(clusterURIString ==~ /hdfs\:\/\/.*?\:[0-9]+/)) {
```

Comment:
I can see this block of code being used multiple times. Can we just abstract out this block of code into a separate function. Since this is just extracting the namenode information, this might be useful for other jobs as well.